core/src/main/java/org/sonarsource/sonarlint/core/util/ProgressReport.java

```diff
@@ -64,19 +65,17 @@ public void message(Supplier<String> messageSupplier) {
 
   public void stop(@Nullable String stopMessage) {
     this.stopMessage = stopMessage;
+    this.stop = true;
     thread.interrupt();
     try {
-      thread.join();
+      thread.join(1000);
     } catch (InterruptedException e) {
       // Ignore
     }
   }
 
   private static void log(String message) {
-    synchronized (LOG) {
-      LOG.info(message);
-      LOG.notifyAll();
-    }
+    LOG.info(message);
```

Comment:
I could not remember/find why this synchronized block was there, so I decided to remove it...

Reply:
Probably because `LOG` can be shared between threads and `LOG.info` might not be thread-safe? E.g if 2 `ProgressReport`s try to log from separate threads.

Reply:
I would be very surprised if LOG is not thread safe. We are often doing concurrent operations (analysis), and so they may log concurrently. This is the only place I know where we are locking like this. Git history shows this was there from the start. I suggest to keep removing it, and if it creates a regression, we will add a test to document why it is there.